compiler/pash_runtime.py

```diff
@@ -686,12 +688,19 @@ def add_eager(eager_input_id, graph, fileIdGen, intermediateFileIdGen, use_dgsh_
     new_id = new_fid.get_ident()
 
     if use_dgsh_tee:
+        assert(False)
         ## TODO: seperate to better use dgsh-tee params and maybe deprecate eager
         eager_node = dgsh_tee.make_dgsh_tee_node(eager_input_id, new_id)
     else:
         ## TODO: Remove the line below if eager creates its intermediate file
         ##       on its own.
+        # TODO: find a better solution to make unique numbers, currently: set to max-value + 1
+        max_old_plus_1 = fileIdGen.next + 1
+        intermediateFileIdGen.next = max_old_plus_1
```

Comment:
Let's move that to a method of FileIdGen that could be called something like "bump_counter/id/next" or something (instead of setting next itself).

Reply:
Ok, I have refactored it.